tsl/src/compression/compression_dml.c

```diff
@@ -498,169 +541,29 @@ decompress_batches_indexscan(Relation in_rel, Relation out_rel, Relation index_r
-
-	while (table_scan_getnextslot(scan, ForwardScanDirection, slot))
+	if (ts_guc_debug_compression_path_info)
 	{
-		num_scanned_rows++;
-		bool skip_tuple = false;
-		int attrno = bms_next_member(null_columns, -1);
-		int pos = 0;
-		bool is_null_condition = 0;
-		bool seg_col_is_null = false;
-		/*
-		 * Since the heap scan API does not support SK_SEARCHNULL we have to check
-		 * for NULL values manually when those are part of the constraints.
-		 */
-		for (; attrno >= 0; attrno = bms_next_member(null_columns, attrno))
-		{
-			/* Treat all conditions as IS NULL if the list is empty */
-			is_null_condition = is_nulls == NIL || list_nth_int(is_nulls, pos);
-			seg_col_is_null = slot_attisnull(slot, attrno);
-			if ((seg_col_is_null && !is_null_condition) || (!seg_col_is_null && is_null_condition))
-			{
-				/*
-				 * if segment by column in the scanned tuple has non null value
-				 * and IS NULL is specified, OR segment by column has null value
-				 * and IS NOT NULL is specified then skip this tuple
-				 */
-				skip_tuple = true;
-				break;
-			}
-			pos++;
-		}
-		if (skip_tuple)
-		{
-			num_filtered_rows++;
-			continue;
-		}
-
-		TM_Result result;
-		Assert(slot->tts_ops->get_heap_tuple);
-		HeapTuple compressed_tuple = slot->tts_ops->get_heap_tuple(slot);
-
-		if (!decompressor_initialized)
-		{
-			decompressor = build_decompressor(in_rel, out_rel);
-			decompressor.delete_only = delete_only;
-			decompressor_initialized = true;
-		}
-
-		heap_deform_tuple(compressed_tuple,
-						  decompressor.in_desc,
-						  decompressor.compressed_datums,
-						  decompressor.compressed_is_nulls);
-
-		if (num_mem_scankeys && !batch_matches(&decompressor,
-											   mem_scankeys,
-											   num_mem_scankeys,
-											   constraints,
-											   skip_current_tuple))
-		{
-			row_decompressor_reset(&decompressor);
-			stats.batches_filtered++;
-			continue;
-		}
-
-		if (skip_current_tuple && *skip_current_tuple)
-		{
-			row_decompressor_close(&decompressor);
-			ExecDropSingleTupleTableSlot(slot);
-			table_endscan(scan);
-			return stats;
-		}
-
-		write_logical_replication_msg_decompression_start();
-		result = delete_compressed_tuple(&decompressor, snapshot, compressed_tuple);
-		/* skip reporting error if isolation level is < Repeatable Read
-		 * since somebody decompressed the data concurrently, we need to take
-		 * that data into account as well when in Read Committed level
-		 */
-		if (result == TM_Deleted && !IsolationUsesXactSnapshot())
-		{
-			write_logical_replication_msg_decompression_end();
-			stats.batches_decompressed++;
-			continue;
-		}
-		if (result != TM_Ok)
-		{
-			table_endscan(scan);
-			report_error(result);
-		}
-		if (decompressor.delete_only)
-		{
-			stats.batches_deleted++;
-		}
+		if (index_rel)
+			elog(INFO,
+				 "Number of compressed rows fetched from index: %d. "
+				 "Number of compressed rows filtered by heap filters: %d.",
+				 num_scanned_rows,
+				 num_filtered_rows);
```

Comment:
```suggestion

			elog(INFO,
				 "Number of compressed rows fetched from %s: %d. "
				 "Number of compressed rows filtered by heap filters: %d.",
				 index_rel ? "index" : "table",
				 num_scanned_rows,
				 num_filtered_rows);
```

Reply:
I made a change along these lines that doesn't change the log output, since it leads to tests breaking. We can consider unifying the log message in both flows with another PR later.